wordpress-mobile/gutenberg-mobile
# Image Block: Improve Text Entry for Long Alt Text

Fixes: #922

`gutenberg`: https://github.com/WordPress/gutenberg/pull/29670

## Description

With this PR, a new `AltTextSettings` component has been introduced to replace the current `TextControl` that exists for entering an image's Alt text. This new component makes use of `BottomSheet.SubSheet` so that users are taken to a new subsheet when entering alt text.

The end goal is to provide a better experience for users who are entering a longer alt text for their images, as the current text entry has limited room/options for scrolling.

## How has this been tested?

The general steps for testing, for both iOS and Android, are as follows:

1. Open **Image Block** settings.
2. Tap **Alt Text** and confirm that a new sub sheet opens with a text editor.
3. Enter text into editor and tap the **back** arrow.
4. Confirm that the image's alt text has been correctly updated by viewing the placeholder displayed alongside **Alt Text** in the settings panel. This can be further verified by saving changes and confirming the alt text has been correctly updated on the web.

As this is a UI change, I have also tested with dark mode enabled and in landscape mode.

Please see Gutenberg PR for additional details, screenshots, and [comments](https://github.com/WordPress/gutenberg/pull/29670#issuecomment-793885334).

PR submission checklist:
